### ISSUE TITLE
script checks wrong value for getRoomUserData

### ISSUE THREAD

**Issue by csmall** [IRE-Mudlet-Mapping/ire-mapping-script#24]
For example, the "Add Room Mark" alias:

```
local tmp = getRoomUserData(1, "gotoMapping")
local maptable = {}

if tmp ~= "" then
  maptable = yajl.to_value(tmp)
end
```

tmp is nil not "", it looks like the wrong check.

**Comment by vadi2** [comment]
I suspect this behaviour was changed - check Mudlet 2.1.

**Comment by vadi2** [comment]
If it was, that'll need to be undone in Mudlet.

**Comment by SlySven** [comment]
Erm, yeah, that was a Mudlet modification :confused: which went into the code base sometime after 3.0.0 "delta" preview and has been/will be "fixed" in Mudlet/Mudlet#340 (development) and Mudlet/Mudlet#341 (release_30, post "epsilon").

The affected versions are the ones that also gain the Map and Areas User Data structures, so you _could_ now move all that storing stuff in room 1 to use:

`setMapUserData( (string)<key>, (string)<value> )`
instead of
`setMapUserData( (number)1, (string)<key>, (string)<value> )`

and:
`getMapUserData( (string)<key> )`
instead of
`getMapUserData( (number)1, (string)<key> )`

though with current preview code you will have to set the "Override map file format version:" option on the "special options" tab of the Profile Preferences ("Settings) dialogue to at least __17__ (the current default is _16_) and the resulting map file will not be loadable by previous versions (they will issue an "`[ ERROR ]  -`" or some other message instead.

It comes about because some time ago (before I started coding for Mudlet BTW) Heiko set the `getRoomUserData(...)` command to return an empty string `""` if the given room number did not exist or the key was not present.  Unfortunately it ___is___ possible to store a empty string as the "value" against a key and in adopting the return a `nil` + `error message` paradigm when I cleaned/tarted-up `getRoomUserData(...)` I did not realise that there was Lua scripts in the wild that were testing for an empty string as proof of the _absence_ of stored data. :disappointed:

The above PRs revert the non-ideal, not able to detect empty but valid `value` so that future versions can run with unmodified scripts and retain backwards compatibility.  Scripts that need to work with the affected 3.0.0 previews could:

* check for existence of `getMudletVersion` if present:
* check for build ="delta" or "epsilon" - if so assume that getRoomUserData(...) will return a _nil_ on no matching key...

After those PRs you will need to furnish an optional _true_ as a third argument to get the more consistent _nil_ + _error message_ behaviour, __which the corresponding area and map user data commands provide from the start!__
